debezium-core/src/test/java/io/debezium/util/Testing.java

```diff
@@ -177,10 +177,33 @@ public static String readResourceAsString(String pathOnClasspath) {
          * @return the reference to the existing readable and writable directory
          */
         public static File createTestingDirectory(String relativePath) {
-            Path dirPath = Paths.get("target/data", relativePath).toAbsolutePath();
+            Path dirPath = createTestingPath(relativePath);
             return IoUtil.createDirectory(dirPath);
         }
 
+        /**
+         * Returns the name of the directory where tests are storing their data. Default value is {@code 'target/data'}.
+         * This value can be overridden by value of the {@code DBZ_TEST_DATA_DIR} system or environment variable.
+         */
+        static String dataDir() {
+            if (DATA_DIR[0] != null) {
+                return DATA_DIR[0];
+            }
+
+            String value = System.getProperty("DBZ_TEST_DATA_DIR");
+            if (value != null && (value = value.trim()).length() > 0) {
+                return DATA_DIR[0] = value;
+            }
+
+            value = System.getenv("DBZ_TEST_DATA_DIR");
+            if (value != null && (value = value.trim()).length() > 0) {
+                return DATA_DIR[0] = value;
+            }
+
+            return DATA_DIR[0] = "target/data"; // default value
+        }
+        String[] DATA_DIR = {null};
```

Comment:
Interesting, why is this an array instead of a single String? I think I'll change it and then apply.

Reply:
@gunnarmorling 
DATA_DIR is final

Reply:
Ok, I've changed it so the variable is initialized eagerly.